slick/src/main/scala/slick/jdbc/SQLiteProfile.scala

```diff
@@ -94,39 +94,48 @@ trait SQLiteProfile extends JdbcProfile {
   )
 
   class ModelBuilder(mTables: Seq[MTable], ignoreInvalidDefaults: Boolean)(implicit ec: ExecutionContext) extends JdbcModelBuilder(mTables, ignoreInvalidDefaults) {
+
     override def createColumnBuilder(tableBuilder: TableBuilder, meta: MColumn): ColumnBuilder = new ColumnBuilder(tableBuilder, meta)
     override def createPrimaryKeyBuilder(tableBuilder: TableBuilder, meta: Seq[MPrimaryKey]): PrimaryKeyBuilder = new PrimaryKeyBuilder(tableBuilder, meta)
 
     class ColumnBuilder(tableBuilder: TableBuilder, meta: MColumn) extends super.ColumnBuilder(tableBuilder, meta) {
-      /** Regex matcher to extract name and length out of a db type name with length ascription */
-      final val TypePattern = "^([A-Z\\s]+)(\\(([0-9]+)\\))?$".r
-      private val (_dbType,_size) = meta.typeName match {
-        case TypePattern(d,_,s) => (d, Option(s).map(_.toInt))
-        case "" => ("TEXT", None)
-      }
+
+      final val TypePattern = "^([A-Z\\s]+)(\\(\\s?([0-9]+)\\s?,?\\s?([0-9]+)?\\s?\\))?$".r
+
+      /** Regex matcher to extract name and length
+       * out of a db type name with length ascription
+       */
+      def extractTypeProps(typeName: String = "NUMERIC(18,2)"): ( String, Option[ Int ] ) =
+        typeName match {
+          case TypePattern(name, _, precision, _) => ( name, Option( precision ).map( _.toInt ) )
+          case "" => ( "TEXT", None )
+        }
+
+      private val ( _dbType, _size ) = extractTypeProps(meta.typeName)
+
       override def dbType = Some(_dbType)
       override def length = _size
       override def varying = dbType == Some("VARCHAR")
       override def default = meta.columnDef.map((_,tpe)).collect{
         case ("null",_)  => Some(None) // 3.7.15-M1
-      	case (v , "java.sql.Timestamp") => {
-      	  import scala.util.{Try, Success}
-      	  val convertors = Seq((s: String) => new java.sql.Timestamp(s.toLong),
+        case (v , "java.sql.Timestamp") => {
+          import scala.util.{Try, Success}
+          val convertors = Seq((s: String) => new java.sql.Timestamp(s.toLong),
             (s: String) => java.sql.Timestamp.valueOf(s),
             (s: String) => new java.sql.Timestamp(javax.xml.bind.DatatypeConverter.parseDateTime(s).getTime.getTime),
-      	    (s: String) => new java.sql.Timestamp(javax.xml.bind.DatatypeConverter.parseDateTime(s.replaceAll(" ","T")).getTime.getTime),
-      	    (s: String) => {
-      		    if(s == "now")
-      		      "new java.sql.Timestamp(java.util.Calendar.getInstance().getTime().getTime())"
-      		    else
-      		      throw new Exception(s"Failed to parse timestamp - $s")
-      	    }
+            (s: String) => new java.sql.Timestamp(javax.xml.bind.DatatypeConverter.parseDateTime(s.replaceAll(" ","T")).getTime.getTime),
```

Comment:
`s.replace(' ', 'T')` or `s.replace(" ", "T")` would also work. The firs argument of replaceAll is a regex, so it requires compiling each time it is executed.